nexus/src/app/sagas/instance_create.rs

```diff
@@ -840,6 +852,48 @@ pub(super) async fn allocate_sled_ipv6(
         .map_err(ActionError::action_failed)
 }
 
+// TODO: Not yet idempotent
```

Comment:
My best idea for idempotency: Couple the "global" resource update with updating the "local" resource.

To elaborate:
- We should create the instance record with cpus = 0, ram = 0
- When we do the accounting operation, we increase them to the desired amount, and update based on the diff
  - This becomes idempotent, because if the operation has previously succeeded, the diff = 0
- We can do the same thing on "delete" (set the resources back to zero before freeing)
- We can *probably* do the same thing for volumes / disk accounting

Reply:
A perk of doing things this way: It also allows us to safely *resize* disks and instances within sagas, when we get around to that